### ISSUE TITLE
idea: Compliance Certification as a solution for Legacy

### ISSUE THREAD

**Issue by refack** [nodejs/TSC#307]
Talking with @chrislea and trying to figure out why users want to run new version of node on old platforms, the issue of system compliance came up. To put it in plain words some organization, "insist" on paying for some sort of Compliance Certification of their systems, and then can not upgrade. Usually it is PCI or some other established certification, but with the rise of node buy-in in many enterprises, maybe the `nodejs.org Certified` brand can be a worthy surrogate.
If this works it achieves three goals:
1. Income
2. Reputation
3. Alleviate legacy support efforts

For Your Consideration

**Comment by Trott** [comment]
@refack I don't think the organizations aren't generally paying for some arbitrary-ish voluntary compliance. They want to pay to not have to worry about compliance issues that they have no choice but to abide by. You can't decide that some other certification is close enough to PCI compliance.

These are the types of compliance I think @chrislea was talking about:

* [PCI DSS](https://en.wikipedia.org/wiki/Payment_Card_Industry_Data_Security_Standard) (often just called "PCI")
* [HIPAA](https://en.wikipedia.org/wiki/Health_Insurance_Portability_and_Accountability_Act)
* [SOX](https://en.wikipedia.org/wiki/Sarbanes%E2%80%93Oxley_Act)
* [FISMA](https://en.wikipedia.org/wiki/Federal_Information_Security_Management_Act_of_2002)
* [GLBA](https://en.wikipedia.org/wiki/Gramm%E2%80%93Leach%E2%80%93Bliley_Act)

...and so on.

For HIPAA and PCI (and I suspect all of these, but those are the two I have the most direct experience with), you can't certify a piece of software or a particular operating system. The certification involves *everything*: The physical security where you stuff is hosted, who has access to it, etc.

@chrislea Please chime in if *I'm* the one whose misunderstood here.

Correction welcome, but I don't think any of this is really feasible. If it were, I'd imagine IBM and/or NodeSource and/or others would already be doing it. (Again, though: Happy to be told I'm wrong if that's the case.)

**Comment by rvagg** [comment]
Obviously I have a bias here because of my association with one of these service providers attempting to work in this space. _But_, one of the reasons I work at NodeSource and am investing my time there is because I genuinely believe that commercial support, certification, someone-to-blame-when-things-go-bad for these big legacy companies that insist on it is an important part of Node's expansion and future health and that can only be supplied by commercially accountable organisations. A non-profit or community-organised effort can't do the same thing, unless it's in the formation of a framework around which commercial entities provide that support. I think there is a role for the Node.js Foundation, and related non-profit corporate collaborations, in providing frameworks, like specs, which is how a lot of these things work anyway.

In terms of the specific issue here, I think we're mostly fine with the existing ecosystem of operating system and Linux distribution vendors. The major commercial players consider Node.js to be essential, witness Microsoft and RedHat's presence at Platinum level in the Foundation. There's also a lot of technical links from those companies and their ecosystems into Node. For example I've been having a conversation with a Canonical group that's working on their new Snaps ecosystem, they're keen to do a good job with Node.js. Many of the kinds of companies that Chris is referring to simply use an off-the-shelf like RHEL, and upgrade rarely because it _just works_. RHEL also has ridiculously long support windows, which is another reason they choose it, it sucks for us, but we have to wear it to support the kinds of companies that can't justify investing in high quality staff to do constant upgrading and maintenance of these systems that are far from essential to their core businesses.

In summary, regarding old systems, I'm not seeing a case for a new framework / certification that's going to improve the situation and it's essential that we accept the fact that Node is going to be deployed in these environments and we need to be somewhat friendly to them if we're not going to turn away masses of users. Think about the poor lone webdev in a big legacy company that wants to do something cutting edge but can't get operations to upgrade from their old RHEL install. There's plenty of them out there and it's to our benefit that we cater to them.

**Comment by refack** [comment]
> @refack I don't think the organizations are generally paying for some arbitrary-ish voluntary compliance. They want to pay to not have to worry about compliance issues that they have no choice but to abide by. You can't decide that some other certification is close enough to PCI compliance.

I was being slightly reductionist. I don't think any of the above certifications are arbitrary pre se, but I do believe that offering a "nodejs stamp of approval" could be a viable addition to this list.
I totally agree that it cannot be a rubber stamp, and that the certification process should have some rigour.

P.S. Having partners (:cough: NodeSource ;cough:) do the actual in-the-field certification, AFAIK, is the common way to do it.

**Comment by mhdawson** [comment]
I think I'm on the same wavelength with @Trott and @rvagg that a new certification would not replace the certifications that cause users to avoid upgrades.

**Comment by jasnell** [comment]
Does this need to remain open?

**Comment by refack** [comment]
I still believe a "Certified by nodejs.org" (provided by in the field partners) would be a good idea, but I don't have the resources to push this forward, so closing.
